src/boundaries/edgetracking.jl

```diff
@@ -0,0 +1,304 @@
+[Skufca2006](@cite), the
+algorithm has been applied to, e.g., the laminar-turbulent boundary in plane Couette
+flow [Schneider2008](@cite), Wada basins [Wagemakers2020](@cite), as well as Melancholia
+states in conceptual [Mehling2023](@cite) and intermediate-complexity [Lucarini2017](@cite) 
+climate models. 
+Relying only on forward integration of the system, it works even in
+high-dimensional systems with complicated fractal basin boundary structures.
+
+The algorithm consists of two main steps: bisection and tracking. First, it iteratively 
+bisects along a straight line in state space between the intial states `u1` and `u2` to find
+the separating basin boundary. The bisection stops when the two updated states are less than
+`bisect_thresh` (Euclidean distance in state space) apart from each other.
+Next, a `ParallelDynamicalSystem` is initialized
+from these two updated states and integrated forward until the two trajectories diverge
+from each other by more than `diverge_thresh` (Euclidean distance). The two final states of
+the parallel integration are then used as new states `u1` and `u2` for a new bisection, and 
+so on, until a stopping criterion is fulfilled. 
+
+Two stopping criteria are implemented via the keyword arguments `maxiter` and `abstol`.
+Either the algorithm stops when the number of iterations reaches `maxiter`, or when the
+state space position of the updated edge point changes by less than `abstol` (in
+Euclidean distance) compared to the previous iteration. Convergence below `abstol` happens
+after sufficient iterations if the edge state is a saddle point. However, the edge state
+may also be an unstable limit cycle or a chaotic saddle. In these cases, the algorithm will
+never actually converge to a point but (after a burn-in) continue populating the set
+consituting the edge state by tracking along it.
+
+A central idea behind this algorithm is that basin boundaries are typically the stable
+manifolds of unstable sets, namely edge states or saddles. The flow along the basin boundary 
+will thus lead to these sets, and the iterative bisection neutralizes the unstable
+direction of the flow away from the basin boundary. If the system possesses multiple edge 
+states, the algorithm will find one of them depending on where the initial bisection locates
+the boundary.
+
+## Output
+
+Returns a data type `EdgeTrackingResults` containing five fields:
+* `edge::StateSpaceSet`: the pseudo-trajectory representing the tracked edge segment
+  (given by the average in state space between `track1` and `track2`)
+* `track1::StateSpaceSet`: the pseudo-trajectory tracking the edge within basin 1
+* `track2::StateSpaceSet`: the pseudo-trajectory tracking the edge within basin 2
+* `time::Vector`: time points of the above `StateSpaceSet`s
+* `bisect_idx::Vector`: indices of `time` at which a re-bisection occurred
+* `success::Bool`: indicates whether the edge tracking has been successful or not
```

Comment:
```suggestion
Returns a data type [`EdgeTrackingResults`](@ref) containing the results.
```

Let's please avoid duplication of information. The type docstring already describes this.